marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/CachePluginConfiguration.kt

```diff
@@ -0,0 +1,71 @@
+package com.malinskiy.marathon
+
+import com.malinskiy.marathon.cache.config.Credentials
+import com.malinskiy.marathon.cache.config.LocalCacheConfiguration
+import com.malinskiy.marathon.cache.config.RemoteCacheConfiguration
+import com.malinskiy.marathon.execution.CacheConfiguration
+import groovy.lang.Closure
+import java.io.File
+import java.lang.IllegalArgumentException
+
+open class CachePluginConfiguration {
+
+    var localExtension: LocalCacheExtension? = null
+    var remoteExtension: RemoteCacheExtension? = null
+
+    fun local(closure: Closure<*>) {
+        localExtension = LocalCacheExtension()
+        closure.delegate = localExtension
+        closure.call()
+    }
+
+    fun remote(closure: Closure<*>) {
+        remoteExtension = RemoteCacheExtension()
+        closure.delegate = remoteExtension
+        closure.call()
+    }
+
+    fun local(block: LocalCacheExtension.() -> Unit) {
+        val config = localExtension ?: LocalCacheExtension()
+        config.also(block)
+        localExtension = config
+    }
+
+    fun remote(block: RemoteCacheExtension.() -> Unit) {
+        val config = remoteExtension ?: RemoteCacheExtension()
+        config.also(block)
+        remoteExtension = config
+    }
+}
+
+private val DEFAULT_LOCAL_CACHE_DIRECTORY = File("~/cache/marathon")
+private const val DEFAULT_LOCAL_UNUSED_ENTRIES_DELETE_AFTER_DAYS = 10
```

Comment:
We can use 7, at least it would be the same as in Gradle https://github.com/gradle/gradle/issues/1085#issuecomment-411986845

Reply:
OK, updated